src/transformer/descriptor/type/type.ts

```diff
@@ -62,5 +58,10 @@ export function GetType(node: ts.Node, scope: Scope): ts.Node {
     return GetType(node.type, scope);
   }
 
+  if (ts.isCallExpression(node)) {
```

Comment:
The responsibility of GetType was to resolve a type given a specific node. For example following imports until you find the type. It is useful to understand upfront if a type is a typescript type or to make sure union types are not trying to combine primitive types. I'm not sure 100% resolving the return type of the function would be in the right place here. I don't mind revisiting the scope of GetTypes.

Reply:
I got it wrong. CallExpression is not like a functionType so it make sense to resolve the return type.